content/docs/extensions/pg_prewarm.md

```diff
@@ -0,0 +1,241 @@
+    - Concurrent modifications: If the table is being modified by inserts, updates, and deletes while the prewarming process is happening or between the size check and the prewarming operation, this could lead to differences. The actual data size in the table might change between the time it's measured and the time `pg_prewarm` runs or completes.
+    - Index data: The example shown above does not include indexes that might be associated with the table. `pg_prewarm` can also prewarm indexes, but if you're only looking at the table's data size, any loaded index data won't be counted.
+    - Overhead and metadata: There is internal overhead and metadata associated with Postgres data storage that might affect calculations.
+
+## Demonstrating the effect of pg_prewarm
+
+This example shows how preloading data can improve query performance. We'll create two tables with the same data, preload one table, and then run `EXPLAIN ANALYZE` to compare execution time results.
+
+1. Create two sample tables with the same data for comparison:
+
+    ```sql
+    CREATE TABLE tbl_transactions_1
+    (
+        tran_id_ SERIAL,
+        transaction_date TIMESTAMPTZ,
+        transaction_name TEXT
+    );
+
+    INSERT INTO tbl_transactions_1
+    (transaction_date, transaction_name)
+    SELECT x, 'dbrnd' 
+    FROM generate_series('2010-01-01 00:00:00'::timestamptz, '2018-02-01 00:00:00'::timestamptz, '1 minutes'::interval) a(x);
+    ```
+
+    ```sql
+    CREATE TABLE tbl_transactions_2
+    (
+        tran_id_ SERIAL,
+        transaction_date TIMESTAMPTZ,
+        transaction_name TEXT
+    );
+
+    INSERT INTO tbl_transactions_2
+    (transaction_date, transaction_name)
+    SELECT x, 'dbrnd' 
+    FROM generate_series('2010-01-01 00:00:00'::timestamptz, '2018-02-01 00:00:00'::timestamptz, '1 minutes'::interval) a(x);
+    ```
+
+2. Restart your Postgres instance to clear the buffer cache. On Neon, you can do this by [restarting your compute](/docs/manage/endpoints#restart-a-compute-endpoint).
+
+2. Prewarm the first sample table:
+
+    ```sql
+    SELECT pg_prewarm('tbl_transactions_1') AS blocks_loaded;
+    ```
+
+    This will output the number of blocks that were loaded into the cache:
+
+    ```sql
+    blocks_loaded 
+    ---------------
+            27805
+    ```
+
+3. Now, compare the execution plan of the prewarmed table vs. a non-prewarmed table to see the performance improvement.
+
+    ```sql
+    EXPLAIN ANALYZE SELECT * FROM tbl_transactions_1;
+    ```
+
+    ```sql
+    EXPLAIN ANALYZE SELECT * FROM tbl_transactions_2;
+    ```
+
+    The execution time for the prewarmed table should be significantly lower than for the table that has not been prewarmed, as shown here:
```

Comment:
I'm not sure what counts as "significantly lower" but this is what I got:

Seq Scan on tbl_transactions_1  (cost=0.00..69608.21 rows=4252321 width=18) (actual time=0.002..188.812 rows=4252321 loops=1)
Planning Time: 0.119 ms
Execution Time: 282.806 ms

Seq Scan on tbl_transactions_2  (cost=0.00..69608.21 rows=4252321 width=18) (actual time=0.003..203.255 rows=4252321 loops=1)
Planning Time: 0.129 ms
Execution Time: 293.606 ms

Reply:
I think stopping the compute to release memory might have been the difference here.